lib/models/api_calls.dart

```diff
@@ -51,24 +51,27 @@ class ApiCalls {
   getDeviceList(String authToken, String nextPageToken) async {
     var response;
     authToken = 'Bearer $authToken';
-    try {
-      nextPageToken == null
-          ? response = await _client.get(DEVICE_LIST_URL,
-              headers: {HttpHeaders.authorizationHeader: authToken})
-          : response = await _client.get(
-              DEVICE_LIST_WITH_TOKEN_URL + nextPageToken,
-              headers: {HttpHeaders.authorizationHeader: authToken});
+    _client == null
```

Comment:
Also for this IDK if you can cache the client you will use for the request, something like:
var clientToUse = _client == null ? http : _client;

Then all you do is clientToUse.get(url, headers: {...})

Reply:
I tried but, http is a prefix to library.